lib/benchee/formatters/console.ex

```diff
@@ -62,27 +62,52 @@ defmodule Benchee.Formatters.Console do
     max_label_width + 1
   end
 
-  defp job_reports(jobs, label_width) do
-    Enum.map(jobs, fn(job) -> format_job job, label_width end)
+  defp job_reports(jobs, units, label_width) do
+    Enum.map(jobs, fn(job) -> format_job job, units, label_width end)
+  end
+
+  defp units(jobs) do
+    # Produces a map like
+    #   %{run_time: [12345, 15431, 13222], ips: [1, 2, 3]}
+    collected_values = jobs
+    |> Enum.flat_map(fn({_name, job}) -> Map.to_list(job) end)
+    # TODO: Simplify when dropping support for 1.2
+    # For compatibility with Elixir 1.2. In 1.3, the following group-reduce-map
+    # can b replaced by a single call to `group_by/3`
+    #   Enum.group_by(fn({measurement, _}) -> measurement end, fn({_, value}) -> value end)
+    |> Enum.group_by(fn({measurement, _unit}) -> measurement end)
+    |> Enum.reduce(%{}, fn({measurement, occurrences}, acc) ->
+      Map.put(acc, measurement, Enum.map(occurrences, fn({_meas, value}) -> value end))
+    end)
+
+    %{
+      run_time: Duration.best(collected_values.average),
+      ips:      Count.best(collected_values.ips),
+    }
   end
 
   defp format_job({name, %{average:       average,
                            ips:           ips,
                            std_dev_ratio: std_dev_ratio,
                            median:        median}
+                         },
+                         %{run_time:      run_time_unit,
+                           ips:           ips_unit,
                          }, label_width) do
-    "~*s~*.2f~*ts~*ts~*ts\n"
-    |> :io_lib.format([-label_width, name, @ips_width, ips,
-                       @average_width, run_time_out(average),
+    "~*s~*ts~*ts~*ts~*ts\n"
+    |> :io_lib.format([-label_width, name, @ips_width, ips_out(ips, ips_unit),
+                       @average_width, run_time_out(average, run_time_unit),
                        @deviation_width, deviation_out(std_dev_ratio),
-                       @median_width, run_time_out(median)])
+                       @median_width, run_time_out(median, run_time_unit)])
     |> to_string
   end
 
-  defp run_time_out(average) do
-    "~.#{float_precision(average)}f~ts"
-    |> :io_lib.format([average, " μs"])
-    |> to_string
+  defp ips_out(ips, unit) do
+    Count.format(Count.scale(ips, unit))
+  end
+
+  defp run_time_out(average, unit) do
+    Duration.format(Duration.scale(average, unit))
```

Comment:
:heart_eyes: how that nitty gritty format code now just lives somewhere else more focussed and contained :)